.travis.yml

```diff
@@ -1,62 +1,103 @@
+jdk:
+- oraclejdk8
 
 env:
-  matrix:
-    - ANDROID_TARGET=android-16 ANDROID_ABI=armeabi-v7a
-    - ANDROID_TARGET=android-19 ANDROID_ABI=armeabi-v7a
-    - ANDROID_TARGET=android-21 ANDROID_ABI=armeabi-v7a
   global:
-    - ADB_INSTALL_TIMEOUT=5 # increase from default of 2 mins
+  - ADB_INSTALL_TIMEOUT=8 # increase from default of 2 mins
+  - ANDROID_EMULATOR=test # default emulator name (used in tests)
+  - ANDROID_ABI=armeabi-v7a
+  - ANDROID_TARGET=android-21
+  - NDK_VERSION=r16b
+  - EMULATOR_FLAVOR=default
+  - ANDROID_HOME=/usr/local/android-sdk
+  - TOOLS=${ANDROID_HOME}/tools
+  - PATH=${ANDROID_HOME}:${ANDROID_HOME}/emulator:${TOOLS}:${TOOLS}/bin:${ANDROID_HOME}/platform-tools:${PATH}
+  - QEMU_AUDIO_DRV=none
+
+android:
+  components:
+  # installing tools to start, then use `sdkmanager` below to get the rest
+  - tools
 
-# see https://github.com/travis-ci/travis-ci/issues/8874
 before_install:
-  - yes | sdkmanager "platforms;android-27"
+- bundle install
+- ./scripts/install-ndk.sh
+- env
+
+install:
+- echo y | sdkmanager "platform-tools" >/dev/null
+- echo y | sdkmanager "tools" >/dev/null
+- echo y | sdkmanager "build-tools;27.0.0" >/dev/null
+- echo y | sdkmanager "platforms;$ANDROID_TARGET" >/dev/null
+# see https://github.com/travis-ci/travis-ci/issues/8874
+- yes | sdkmanager "platforms;android-27" >/dev/null
+- echo y | sdkmanager "extras;android;m2repository" >/dev/null
+- echo y | sdkmanager "system-images;$ANDROID_TARGET;$EMULATOR_FLAVOR;$ANDROID_ABI" >/dev/null
+- echo y | sdkmanager "platforms;$ANDROID_TARGET" >/dev/null
+- echo no | avdmanager create avd --force -n test -k "system-images;$ANDROID_TARGET;$EMULATOR_FLAVOR;$ANDROID_ABI" -c 10M
 
 before_script:
-  # Create and start emulator
-  - android list targets
-  - jdk_switcher use oraclejdk8
-  - echo no | android create avd --force -n test -t $ANDROID_TARGET --abi $ANDROID_ABI
-  - emulator -avd test -no-skin -no-audio -no-window &
-  - adb wait-for-device
-  - while [[ `adb shell pm path android` == 'Error'* ]]; do sleep 2; done
-  - adb shell input keyevent 82 &
+- emulator -avd test $EMULATOR_FLAGS -engine classic -no-window -camera-back none -camera-front none -selinux permissive &
+- android-wait-for-emulator
+- adb shell input keyevent 82 &
 
 script:
-- ./gradlew lint checkstyle createDebugCoverageReport coveralls
-- cd examples/sdk-app-example && ../../gradlew lint checkstyle connectedCheck
+- bundle exec bugsnag-maze-runner --color features/native*.feature --verbose
 
-sudo: false
+jobs:
+  include:
+  - stage: lint
+    script: ./gradlew lint checkstyle
+  - stage: unit tests
+    script: ./gradlew createDebugCoverageReport coveralls || (adb logcat -v brief -d '*:S BugsnagNDKTest' && false)
+    env: NDK_VERSION=r14b
+  - stage: unit tests
+    script: ./gradlew createDebugCoverageReport coveralls || (adb logcat -v brief -d '*:S BugsnagNDKTest' && false)
+    env: NDK_VERSION=r16b
+  # Emulators for API 22,23,25 just don't boot ¯\_(ツ)_/¯
```

Comment:
This is probably down to a change in how the Android Device images are named - after a certain API level, the format `system-images;$ANDROID_TARGET;$EMULATOR_FLAVOR;$ANDROID_ABI` can't be relied upon. However, I've never seen a working example of an Android emulator >API 24 running on Travis CI, so it's probably not worth attempting to setup as part of this changeset.

Reply:
> However, I've never seen a working example of an Android emulator >API 24 running on Travis CI

This was the core issue. Even if I change the invocations to packages that work locally for API >24, the emulator installs but still doesn’t boot, hence the comment.